adafruit_sht31.py

```
@@ -0,0 +1,144 @@
+# THE SOFTWARE.
+"""
+`adafruit_sht31`
+====================================================
+
+This is a CircuitPython driver for the SHT31-D temperature and humidity sensor.
+
+"""
+
+# imports
+try:
+    import struct
+except ImportError:
+    import ustruct as struct
+
+import time
+
+from adafruit_bus_device.i2c_device import I2CDevice
+from micropython import const
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_sht31.git"
+
+
+SHT31_DEFAULT_ADDR = const(0x44)
+SHT31_MEAS_HIGHREP_STRETCH = const(0x2C06)
+SHT31_MEAS_MEDREP_STRETCH = const(0x2C0D)
+SHT31_MEAS_LOWREP_STRETCH = const(0x2C10)
+SHT31_MEAS_HIGHREP = const(0x2400)
+SHT31_MEAS_MEDREP = const(0x240B)
+SHT31_MEAS_LOWREP = const(0x2416)
+SHT31_READSTATUS = const(0xF32D)
+SHT31_CLEARSTATUS = const(0x3041)
+SHT31_SOFTRESET = const(0x30A2)
+SHT31_HEATEREN = const(0x306D)
+SHT31_HEATERDIS = const(0x3066)
+
+
+def _crc(data):
+    crc = 0xff
+    for byte in data:
+        crc ^= byte
+        for _ in range(8):
+            if crc & 0x80:
+                crc <<= 1
+                crc ^= 0x131
+            else:
+                crc <<= 1
+    return crc
+
+
+class SHT31:
+    """
+    A driver for the SHT31-D temperature and humidity sensor.
+    """
+    def __init__(self, i2c_bus, address=SHT31_DEFAULT_ADDR):
+        self.i2c_device = I2CDevice(i2c_bus, address)
+        self._command(SHT31_SOFTRESET)
+        time.sleep(.010)
+
+    def _command(self, command):
+        with self.i2c_device as i2c:
+            i2c.write(struct.pack('>H', command))
+
+    def _data(self):
+        data = bytearray(6)
+        data[0] = 0xff
+        self._command(SHT31_MEAS_HIGHREP)
+        time.sleep(.5)
+        try:
+            with self.i2c_device as i2c:
+                i2c.readinto(data)
+        except OSError as error:
+            if error.args[0] not in ('I2C bus error', 19):  # errno 19 ENODEV
+                raise
```

Why hide the error that no device is found?

---

This was a hold over from the si7021 - It should not be here - I'll take it out.

---

I don't think you need the try except at all now. You are just re-raising it.